tests/baselines/reference/optionalMethods.js

```diff
@@ -138,12 +138,12 @@ interface Foo {
 }
 declare function test1(x: Foo): void;
 declare class Bar {
-    d?: number | undefined;
+    d?: number;

```

Comment:
I am somewhat suspicious of the class properties just because of exactOptionalPropertyType; the other PR went out of its way to add `| undefined` back, but maybe this doesn't matter in practice given the code that generates it doesn't have the flag on so produces code for that case (sorta like if you compile without strictNullChecks... you get weird stuff).

Reply:
As does this PR... for required initialized parameters. Properties, however, retain their `?`, and only even "do the right thing" under `exactOptionalPropertyTypes` when you *don't* print an `undefined` unless the user wrote it (and, under any other mode, it doesn't matter). We've been super loose and just print whatever type we get back (for odd reasons), while now we go "oh, this is optional, a `number` here means the same thing as the `number | undefined` we got back from the checker, so we can just reuse the `number` annotation". Notably, "reuse the annotation" is our normal behavior everywhere else (because this is definitely annotated) - it's actually a little confusing why we weren't retaining it here before.